gapir/cc/memory_allocator_test.cpp

```diff
@@ -0,0 +1,295 @@
+
+  for (unsigned int i = 0; i < addresses.size(); ++i) {
+    if (i % 5 == 0 || i % 7 == 0 || i % 11 == 0) {
+      EXPECT_TRUE(allocator->releaseAllocation(addresses[i]));
+      addresses[i] = MemoryAllocator::Handle();
+      expectPass[i] = false;
+    }
+  }
+
+  TEST_ALLOCATIONS(addresses, allocationSizes, expectPass);
+
+  for (unsigned int i = 0; i < 98; ++i) {
+    allocationSizes.push_back(8);
+    expectPass.push_back(i < 97);
+
+    auto alloc = allocator->allocatePurgable(8);
+    addresses.push_back(alloc);
+  }
+
+  TEST_ALLOCATIONS(addresses, allocationSizes, expectPass);
+
+  for (unsigned int i = 257; i < addresses.size(); ++i) {
+    if (i % 13 == 0 || i % 17 == 0 || i % 19 == 0) {
+      EXPECT_TRUE(allocator->releaseAllocation(addresses[i]));
+      addresses[i] = MemoryAllocator::Handle();
+      expectPass[i] = false;
+    }
+  }
+
+  TEST_ALLOCATIONS(addresses, allocationSizes, expectPass);
+
+  for (unsigned int i = 0; i < 18; ++i) {
+    allocationSizes.push_back(8);
+    expectPass.push_back(i < 17);
+
+    auto alloc = allocator->allocatePurgable(8);
+    addresses.push_back(alloc);
+  }
+
+  TEST_ALLOCATIONS(addresses, allocationSizes, expectPass);
+}
+
+TEST(MemoryAllocator, SimplePurgableAllocatePurge) {
+  std::unique_ptr<MemoryAllocator> allocator = MemoryAllocator::create(2048);
+
+  std::vector<size_t> allocationSizes;
+  std::vector<bool> expectPass;
+
+  std::vector<MemoryAllocator::Handle> addresses;
+
+  for (unsigned int i = 0; i < 256; ++i) {
+    allocationSizes.push_back(8);
+    expectPass.push_back(i < 256);
+
+    auto alloc = allocator->allocatePurgable(8);
+    addresses.push_back(alloc);
+  }
+
+  TEST_ALLOCATIONS(addresses, allocationSizes, expectPass);
+
+  EXPECT_NE(allocator->allocateStatic(1024), nullptr);
+
+  unsigned int purgedAllocations = 0;
+  for (auto&& alloc : addresses) {
+    if (alloc == nullptr) {
+      purgedAllocations++;
+    }
+  }
+
+  EXPECT_EQ(128, purgedAllocations);
+
+  allocator->releaseAllocation(addresses[0]);
+  addresses.push_back(allocator->allocatePurgable(8));
+  allocationSizes.push_back(8);
+  expectPass.push_back(true);
+
+  purgedAllocations = 0;
+  for (auto&& alloc : addresses) {
+    if (alloc == nullptr) {
+      purgedAllocations++;
+    }
+  }
+
+  EXPECT_EQ(129, purgedAllocations);
+}
+
+TEST(MemoryAllocator, SimplePurgableAllocatePurgeViaGrow) {
+  std::unique_ptr<MemoryAllocator> allocator = MemoryAllocator::create(2048);
+
+  std::vector<size_t> allocationSizes;
+  std::vector<bool> expectPass;
+
+  std::vector<MemoryAllocator::Handle> addresses;
+
+  for (unsigned int i = 0; i < 256; ++i) {
+    allocationSizes.push_back(8);
+    expectPass.push_back(i < 256);
+
+    auto alloc = allocator->allocatePurgable(8);
+    addresses.push_back(alloc);
+  }
+
+  auto alloc = allocator->allocateStatic(1024);
+  EXPECT_NE(alloc, nullptr);
+
+  allocator->resizeStaticAllocation(alloc, 1536);
+
+  allocator->resizeStaticAllocation(alloc, 1024);
+}
```

Comment:
Add a test to create a memory allocator with a size so big that we can expect the creation to fail?
(To test the logic of cmd/gapir/cc/main.cpp:67)